File: Algorithm/c_or_cpp/charpter9_binary/level2/peak_index_in_mountain_array.cpp
#include <stdio.h>

 
int peakIndexInMountainArray(int* arr, int arrSize) {
    int n = arrSize;
    int ans = -1;
    for (int i = 1; i < n - 1; ++i) {
        if (arr[i] > arr[i + 1]) {
            ans = i;
            break;
        }
    }
    return ans;
}

int main()
{
    int array[] = {1, 2, 3, 4, 3, 2, 1};
    int index = peakIndexInMountainArray(array, 7);
    if (index == -1)
    {
        printf("Target not found in the array.\n");
    }
    else
    {
        printf("Target found at index %d.\n", index);
    }
    return 0;
}

File: Algorithm/Code_Materials/Java/src/main/java/com/yugutou/charpter9_bitree_bisearch/level2/PeakIndexInMountainArray.java
package com.yugutou.charpter9_bitree_bisearch.level2;

/**
 * LeetCode852.山峰数组，
 * 在数组中的某位位置i开始，从0到i是递增的，从i+1 到数组最后是递减的，让你找到这个最高点。
 */
public class PeakIndexInMountainArray {
    public static void main(String[] args) {
        int[] arr = {0, 1, 0};
        System.out.println(peakIndexInMountainArray(arr));
    }

    public static int peakIndexInMountainArray(int[] arr) {
        int n = arr.length;
        int ans = -1;
        for (int i = 1; i < n - 1; ++i) {
            if (arr[i] > arr[i + 1]) {
                ans = i;
                break;
            }
        }
        return ans;
    }
}
